### PR TITLE
[LW-7681] lace-blockchain-services: HTTP API to control the network switch and view available services

### PR DESCRIPTION
**Pull request by michalrus**
# Checklist

- [ ] JIRA - https://input-output.atlassian.net/browse/LW-7681
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Define a few API endpoints to control the network switch, and service discovery.

## Testing

The following should work the same **on all 3 platforms** (4, if you count macOS on Apple Silicon separately).

1. From the system tray menu, select "**OpenAPI Specification v3.0**", and click around.

    1. Submit a `/v1/switch-network` from the Swagger UI ("Try it out" → "Execute"). It should switch the network (same effect as selecting that from the system tray UI). Check logs, the tray menu, and the WebSocket output.

    1. Check if the `/v1/info` endpoint is responding with reasonable data (compared to what's displayed in the tray menu).

1. From the system tray menu, select "**Dashboard**", and click around.

    1. In there, watch what's being written to the WebSocket (`/v1/websocket`) when you switch networks, watch sync progress etc.

    1. Try switching networks by pressing the three NetworkMagic buttons on the left side (see the recording below).

## Screenshots

<table>
<tr>
<td>

</td>
<td>

</td>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
</table>

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1297/5750546822/index.html) for [9c26fbe4](https://github.com/input-output-hk/lace/pull/319/commits/9c26fbe445a5b2dcb68a24b30fa4ac0f368bf897)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 3      | 0       | 0     | 36    | ❌     |

### CODE REVIEW

File: nix/lace-blockchain-services/internal/any-darwin.nix [hunk 1]
```diff
@@ -98,7 +98,7 @@ in rec {
   lace-blockchain-services-exe = pkgs.buildGoModule rec {
     name = "lace-blockchain-services";
     src = ./lace-blockchain-services;
-    vendorHash = "sha256-1slTIiIGxraIFdtKNeH4llXjrtSEaEQ7IIbOM3LL3N0=";
+    vendorHash = common.ourVendorHash;
```
Comment (gytis-ivaskevicius) [inline nix/lace-blockchain-services/internal/any-darwin.nix:101]: **☭**
I am not a fan of the variable name, not sure what it should be but maybe we could do better?

**Review by gytis-ivaskevicius — APPROVED** [review]
I would be curious to hear more about these lace-blockchain-services. It is getting awfully complicated and I am wondering maybe there is an easier solution to all of this :thinking:

**Comment by michalrus** [comment]
> It is getting awfully complicated 

I hope it shouldn’t get much more complicated :eyes: A few more child processes, but that is already abstracted away.

> and I am wondering maybe there is an easier solution to all of this thinking For starters maybe we could use some job scheduler to simplify things?

It would have to work on all three platforms, send correct "signals" on Windows (cardano-node is a bit non-standard here (but clever) in that it requires inherited FDs), be able to handle sync progress from Ogmios instead of cardano-node. 

A lot of custom tweaks, and the current code is not that big:

```
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Go                              18            288             88           1803
Nix                              7            199            334           1071
HTML                             1             42              0            257
JSON                             1              0              0            199
Haskell                          1             30             11            184
[…]
```

Esp. since the HTML parts will probably be scrapped, and we’ll have something from our proper JS devs

File: nix/lace-blockchain-services/internal/any-darwin.nix (hunk 1 above)
Comment (michalrus) [reply nix/lace-blockchain-services/internal/any-darwin.nix:101]: Ah, you only meant variable name – ☭! :stuck_out_tongue:  Sorry

Hmmm, `lace-blockchain-services-exe-vendorHash`? 🫣

Comment (rhyslbw) [reply nix/lace-blockchain-services/internal/any-darwin.nix:101]: What's the outcome here @michalrus ?

**Review by rhyslbw — COMMENTED** [review]
Awesome work @michalrus 🧙 

Maybe a more user-centric name, rather than Websocket UI, is _Monitoring Dashboard_ or just simple _Dashboard_?

**Comment by michalrus** [comment]
Thank you! :bow: 

Okay, I’ll change it :hourglass:

**Comment by michalrus** [comment]
@rhyslbw done in 0b1656741e39d06073bfdfd74065b21e6d9208c8 :slightly_smiling_face:

**Review by rhyslbw — APPROVED** [review]
Excellent work @michalrus 🚀

File: nix/lace-blockchain-services/internal/any-darwin.nix (hunk 1 above)
Comment (gytis-ivaskevicius) [reply nix/lace-blockchain-services/internal/any-darwin.nix:101]: I would go with something shorter like `blockchainServicesVendorHash` but I am okay with your proposed name

**Pull request merged**